astroquery/heasarc/tests/test_heasarc.py

```diff
@@ -1,2 +1,82 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+import os
+
 from __future__ import print_function
+from astropy.tests.helper import pytest, catch_warnings
+from astropy.utils.exceptions import AstropyWarning
+
+from ...utils.testing_tools import MockResponse
+from ..heasarc import HeasarcClass
+
+data_filenames = {"good": "testing_table.fits",
+                  "bad_table": "bad_table.html",
+                  "bad_object": "bad_object.html",
+                  "fallback": "table_fallback.fits"}
+
+
+# some basic defs of names
+good_object = "3c273"
+good_table = "rospublic"
+bad_object = "ImABadObject"
+bad_table = "ImABadTable"
+fallback_table = "numaster"
+
+def get_mockreturn(url, params=None, timeout=10, **kwargs):
+    obejct_name = params.get('object_name')
+    table_name = params.get('mission')
+
+    if object_name == good_object and table_name == good_table:
+        content = open(data_path(data_filenames['good']), "rb").read()
+    elif object_name == bad_object:
+        content = open(data_path(data_filenames['bad_object']), "r").read()
+    elif table_name == bad_table:
+        content = open(data_path(data_filenames['bad_table']), "r").read()
+    elif table_name == fallback_table:
+        content = open(data_path(data_filenames['fallback']), "rb").read()
+
+    return MockResponse(content, **kwargs)
+
+
+def data_path(filename):
+    data_dir = os.path.join(os.path.dirname(__file__), 'data')
+    return os.path.join(data_dir, filename)
+
+
+@pytest.fixture
+def patch_get(request):
+    mp = request.getfuncargvalue("monkeypatch")
+    mp.setattr(requests, 'get', get_mockreturn)
```

Comment:
this should patch ``Heasarc._request`` and not ``requests.get``, for examples see recent PRs #821 #822